subsys/bluetooth/audio/bass_client.c

```diff
@@ -0,0 +1,868 @@
+	bass_client.write_params.data = buf->data;
+	bass_client.write_params.length = buf->len;
+	bass_client.write_params.handle = bass_client.cp_handle;
+	bass_client.write_params.func = bass_client_write_cp_cb;
+
+	err = bt_gatt_write(conn, &bass_client.write_params);
+	if (err == 0) {
+		bass_client.busy = true;
+	}
+
+	return err;
+}
+
+
+static bool broadcast_source_found(struct bt_data *data, void *user_data)
+{
+	const struct bt_le_scan_recv_info *info = user_data;
+	struct bt_uuid_16 adv_uuid;
+	uint32_t broadcast_id;
+
+	if (data->type != BT_DATA_SVC_DATA16) {
+		return true;
+	}
+
+	if (data->data_len < BT_UUID_SIZE_16 + BT_BASS_BROADCAST_ID_SIZE) {
+		return true;
+	}
+
+	if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
+		return true;
+	}
+
+	if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO) != 0) {
+		return true;
+	}
+
+	broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+
+	BT_DBG("Found BIS advertiser with address %s",
+	       bt_addr_le_str(info->addr));
+
+	if (bass_cbs != NULL && bass_cbs->scan != NULL) {
+		bass_cbs->scan(info, broadcast_id);
+	}
+
+	return false;
+}
+
+void scan_recv(const struct bt_le_scan_recv_info *info,
+	       struct net_buf_simple *ad)
+{
+	/* We are only interested in non-connectable periodic advertisers */
+	if ((info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) != 0 ||
+	    info->interval == 0) {
+		return;
+	}
+
+	bt_data_parse(ad, broadcast_source_found, (void *)info);
+}
+
+static struct bt_le_scan_cb scan_cb = {
+	.recv = scan_recv
+};
+
+/****************************** PUBLIC API ******************************/
+
+int bt_bass_client_discover(struct bt_conn *conn)
+{
+	int err;
+
+	if (conn == NULL) {
+		return -EINVAL;
+	}
+
+	/* Discover BASS on peer, setup handles and notify */
+	(void)memset(&bass_client, 0, sizeof(bass_client));
+	(void)memcpy(&uuid, BT_UUID_BASS, sizeof(uuid));
+	bass_client.disc_params.func = primary_char_discover_func;
+	bass_client.disc_params.uuid = &uuid.uuid;
+	bass_client.disc_params.type = BT_GATT_DISCOVER_PRIMARY;
+	bass_client.disc_params.start_handle = BT_ATT_FIRST_ATTRIBUTE_HANDLE;
+	bass_client.disc_params.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE;
+
+	err = bt_gatt_discover(conn, &bass_client.disc_params);
+	if (err != 0) {
+		return err;
+	}
+
+	bass_client.discovering = true;
+
+	return 0;
+}
+
+void bt_bass_client_register_cb(struct bt_bass_client_cb *cb)
+{
+	bass_cbs = cb;
+}
+
+int bt_bass_client_scan_start(struct bt_conn *conn, bool start_scan)
+{
+	struct bt_bass_cp_scan_start *cp;
+	int err;
+
+	if (conn == NULL) {
+		return -EINVAL;
+	} else if (bass_client.cp_handle == 0) {
+		return -EINVAL;
+	} else if (bass_client.busy) {
+		return -EBUSY;
+	}
+
+	if (start_scan) {
+		static bool cb_registered;
+
+		if (!cb_registered) {
+			bt_le_scan_cb_register(&scan_cb);
+			cb_registered = true;
+		}
+
+		err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, NULL);
+		if (err != 0) {
+			BT_DBG("Could not start scan (%d)", err);
+
```

Comment:
nitpick: Isn't this newline unnecessary?

Reply:
Not for `BT_DBG` :)